Softeq.NetKit.Integrations.EventLog/IntegrationEventLogService.cs

```diff
@@ -1,113 +1,255 @@
 using System.Threading.Tasks;
+using Softeq.NetKit.Integrations.EventLog.Dtos;
+using SortOrder = Softeq.NetKit.Integrations.EventLog.Dtos.SortOrder;
 
 namespace Softeq.NetKit.Integrations.EventLog
 {
     public class IntegrationEventLogService : IIntegrationEventLogService
     {
-        protected IntegrationEventLogContext EventLogContext;
+        private readonly Func<IntegrationEventLogContext> _dbContextFactory;
 
-        public IntegrationEventLogService(IntegrationEventLogContext eventLogContext)
+        protected IntegrationEventLogContext DbContext => _dbContextFactory.Invoke();
+
+        public IntegrationEventLogService(Func<IntegrationEventLogContext> dbContextFactory)
         {
-            EventLogContext = eventLogContext;
+            _dbContextFactory = Ensure.Any.IsNotNull(dbContextFactory, nameof(dbContextFactory));
         }
 
+        /// <inheritdoc />
         public async Task<IntegrationEventLog> GetAsync(Guid eventId)
         {
-            var eventLog = await EventLogContext.IntegrationEventLogs.FirstOrDefaultAsync(log => log.EventId == eventId);
+            Ensure.Guid.IsNotEmpty(eventId, nameof(eventId));
+
+            var eventLog = await DbContext
+                .IntegrationEventLogs
+                .FirstOrDefaultAsync(log => log.EventId == eventId);
             if (eventLog == null)
             {
                 throw new EventLogNotFoundException(eventId);
             }
-
             return eventLog;
         }
 
-        public async Task<List<IntegrationEventLog>> GetAsync(Expression<Func<IntegrationEventLog, bool>> condition)
+        /// <inheritdoc />
+        public Task<List<IntegrationEventLog>> GetAsync(Expression<Func<IntegrationEventLog, bool>> condition)
         {
-            if (condition == null)
+            Ensure.Any.IsNotNull(condition, nameof(condition));
+
+            return DbContext.IntegrationEventLogs.Where(condition).ToListAsync();
+        }
+
+        /// <inheritdoc />
+        public Task<List<IntegrationEventContentDto>> GetEventContentListAsync(
+            List<EventState> eventStates,
+            DateTimeOffset createdUntil,
+            SortOrder createdDateOrder = SortOrder.Ascending,
+            int skipCount = 0,
+            int takeCount = 100,
+            CancellationToken cancellationToken = default)
+        {
+            Ensure.Any.IsNotNull(eventStates, nameof(eventStates));
+            Ensure.Collection.HasItems(eventStates, nameof(eventStates));
+            Ensure.Comparable.IsGte(skipCount, 0, nameof(skipCount));
+            Ensure.Comparable.IsGt(takeCount, 0, nameof(takeCount));
+
+            var stateParamNames = eventStates.Select((_, i) => $"@state{i}").ToList();
+            var eventStateInClause = string.Join(", ", stateParamNames);
+            var orderClause = createdDateOrder == SortOrder.Ascending ? "ASC" : "DESC";
+
+            var sql = $@"
+                SELECT [EventId], [EventTypeName], [Content], [Created]
```

Comment:
It seems that there are no benefits from using raw SQL (in terms of performance or specific SQL maintenance queries). Does it make sense to use EF Core?